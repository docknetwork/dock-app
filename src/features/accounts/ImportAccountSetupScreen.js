import React, {useState} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {addTestId} from 'src/core/automation-utils';
import {
  Content,
  Footer,
  Header,
  Input,
  LoadingButton,
  NBox as Box,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {BackButton} from '../../design-system/buttons';
import {translate} from '../../locales';
import {
  createAccountOperations,
  createAccountSelectors,
} from '../account-creation/create-account-slice';

export function ImportAccountSetupScreen({
  form,
  onChange,
  submitDisabled,
  onSubmit,
}) {
  return (
    <ScreenContainer {...addTestId('ImportAccountSetupScreen')}>
      <Header>
        <BackButton {...addTestId('BackButton')} />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('import_account.title')}
        </Typography>
        <Box mt={7}>
          <Input
            placeholder={translate('import_account.account_name_input')}
            value={form.accountName}
            onChangeText={onChange('accountName')}
            autoCapitalize="none"
            {...addTestId('ImportAccountNameInput')}
          />
        </Box>
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <LoadingButton
          {...addTestId('NextBtn')}
          full
          isDisabled={submitDisabled}
          onPress={onSubmit}>
          {translate('navigation.next')}
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}

export function ImportAccountSetupContainer() {
  const dispatch = useDispatch();
  const currentForm = useSelector(createAccountSelectors.getForm);
  const [form, setForm] = useState({
    ...currentForm,
    accountName: currentForm.accountName || '',
    _errors: {},
    _hasError: false,
  });

  const handleChange = key => evt => {
    setForm(v => ({
      ...v,
      [key]: evt,
    }));
  };

  const handleSubmit = () => {
    return dispatch(
      createAccountOperations.createAccount({
        hasBackup: true,
        successMessage: translate('import_account.import_success'),
        form,
      }),
    );
  };

  const submitDisabled = !form.accountName;

  return (
    <ImportAccountSetupScreen
      form={form}
      submitDisabled={submitDisabled}
      onChange={handleChange}
      onSubmit={handleSubmit}
    />
  );
}
