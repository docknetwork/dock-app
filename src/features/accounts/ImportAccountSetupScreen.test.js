import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ImportAccountSetupScreen} from './ImportAccountSetupScreen';

const mockStore = configureMockStore();

describe('ImportAccountSetupScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(
      <ImportAccountSetupScreen
        onChange={jest.fn()}
        form={{
          accountName: 'test'
        }}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
