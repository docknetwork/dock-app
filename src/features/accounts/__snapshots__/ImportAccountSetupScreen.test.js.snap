// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportAccountSetupScreen should render correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  background-color: ",
          [Function],
          ";
  flex: 1;
",
        ],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  testID="create-wallet-screen"
>
  <Styled(View)>
    <BackButton />
  </Styled(View)>
  <Styled(ScrollView)
    marginLeft={26}
    marginRight={26}
  >
    <Styled(Component)
      marginTop={52}
      variant="h1"
    >
      Account name
    </Styled(Component)>
    <Memo(ForwardRef(Box))
      mt={7}
    >
      <Memo(ForwardRef(Input))
        autoCapitalize="none"
        placeholder="Account name"
        value="test"
      />
    </Memo(ForwardRef(Box))>
  </Styled(ScrollView)>
  <Styled(Box)
    marginBottom={10}
    marginLeft={26}
    marginRight={26}
  >
    <LoadingButton
      full={true}
      testID="next-btn"
    >
      Next
    </LoadingButton>
  </Styled(Box)>
</StyledNativeComponent>
`;
